src/app/components/todos/todos.component.ts

```diff
@@ -20,14 +20,14 @@ export class TodosComponent implements OnInit {
   }
 
   deleteTodo(todo:Todo){
-    this.todos = this.todos.filter(t=>t.id!==todo.id);
-    this.todoService.deleteTodo(todo).subscribe();
+    this.todoService.addTodo(todo).subscribe(todo => {
+      this.todos.push(todo);
+    );
   }
 
   addTodo(todo:Todo){
-    this.todoService.addTodo(todo).subscribe(todo => {
-      this.todos.push(todo);
-    });
+    this.todos == this.todos.filter(t=>t.id!==todo.id);
```

Comment:
Use the assignment operator `=` instead of the comparison operator `==` to update `this.todos` correctly.